rust/src/util/api.rs

```diff
@@ -67,10 +82,13 @@ pub fn init_log_fd(log_fd: libc::c_int) -> repr_c::Box<InitLogFdResponse> {
 
 #[cfg(test)]
 mod tests {
-
+    #[cfg(any(feature = "opencl", feature = "cuda"))]
     use crate::util::api::get_gpu_devices;
+
+    #[cfg(any(feature = "opencl", feature = "cuda"))]
     use crate::util::types::destroy_gpu_device_response;
 
+    #[cfg(any(feature = "opencl", feature = "cuda"))]
```

Comment:
In case you don't want to repeat the feature flags, you can also do the imports directly in the test function. So it would look like:


```rust
    #[cfg(any(feature = "opencl", feature = "cuda"))]
    #[test]
    #[allow(clippy::needless_collect)]
    fn test_get_gpu_devices() {
        use crate::util::api::get_gpu_devices;
        use crate::util::types::destroy_gpu_device_response;

        let resp = get_gpu_devices();
        …
    }
```